### PR TITLE
fix: added rounded border to activity card and less opacity when disa…

### PR DESCRIPTION
**Pull request by adeiji**
…bled

### 📝 Description

The corners of the activity cards lost their rounded corners during the RN upgrade. This PR adds the rounded corner styling back, as well as the .5 opacity when the activity card is disabled.

🔗 [Jira Ticket M2-#11117](https://mindlogger.atlassian.net/browse/M2-11117)

### 📸 Screenshots

Before
card corner issue
------------------------- | ------------------------------------- |

After                                 
Screenshot 2026-09-08 at 11 00 21 AM

### Functionality

- [ ] The feature behaves correctly in practice and fulfills the intended business purpose
- [ ] The implementation accounts for edge cases, avoids subtle logical errors, and handles somewhat rare failure states (e.g. offline mode for mobile, 3rd party being down, etc)

### Testing

- [ ] Verify there are automated tests added that meaningfully cover critical behavior and failure cases
- [ ] Code coverage does not go down as result of this change
- [ ] Test suite passes

### Security & Data Privacy

- [ ] Verify there is no chance we would accidentally log PII to application logs
- [ ] Verify this addition does not materially affect our security attack surface, and if so it has undergone security review
- [ ] All inputs are sanitized
- [ ] New dependencies are well maintained, have significant justification for being added to the project, and are documented in the Curious [open source credit page](https://mindlogger.atlassian.net/jira/servicedesk/projects/MLA/knowledge/articles/340623543?spaceKey=MLA)

### Logging/Monitoring

- [ ] Logging is implemented for this change such that you could troubleshoot this feature in production
- [ ] The change/feature is able to be monitored in production

### Performance

- [ ] This change does not introduce n+1 queries or other performance issues within our expected scale (e.g. missing indexes on frequently queried columns, frequently updating tables that are accessed often)

### Readability

- [ ] All commented out code is removed
- [ ] Debugging code including extraneous log lines are removed
- [ ] Code is easy to understand through naming and structure; comments explain intent or non‑obvious decisions

### Change Safety

- [x] ~~Backend changes are backwards compatible with old clients, or it is well known they are not and a deployment/rollout plan is in place. This include backend changes being compatible with old mobile app versions, as well as applet versioning within Curious.~~
- [x] ~~Destructive database migrations are rolled out in stages. For example, renaming a column means adding a new column and migrating the existing data to that columns in one deployment. Then monitoring to ensure that field isn’t used, and finally removing that old column in a separate deployment.~~

### CODE REVIEW

File: src/entities/activity/ui/ActivityCard.tsx [hunk 1]
```diff
@@ -57,6 +57,10 @@ export const ActivityCard: FC<Props> = ({
       aria-label={accessibilityLabel}
       onPress={onPress}
       disabled={isDisabled}
+      style={{
+        borderRadius: 16,
+        opacity: disabled ? 0.5 : 1,
```
Comment (divbzero) [inline src/entities/activity/ui/ActivityCard.tsx]: Should we be using `isDisabled` here as well?

File: src/entities/activity/ui/ActivityCard.tsx
Comment (divbzero) [inline src/entities/activity/ui/ActivityCard.tsx:66]: How does `<YStack opacity={isDisabled ? 0.5 : 1}>` interact with `<AnimatedTouchable style={{ opacity: disabled ? 0.5 : 1 }}>` above? Maybe we just need one of these?

**Review by divbzero — APPROVED** [review]
@adeiji Maybe one quick fix on `opacity` but otherwise looks good!

File: src/entities/activity/ui/ActivityCard.tsx
Comment (adeiji) [reply src/entities/activity/ui/ActivityCard.tsx:66]: Good catch, done.

File: src/entities/activity/ui/ActivityCard.tsx (hunk 1 above)
Comment (adeiji) [reply src/entities/activity/ui/ActivityCard.tsx]: Removed according to comment below.

**Comment by divbzero** [comment]
Rebased against `fix/activity-status-bar-jank-m2-10985` to resolve conflicts in:
- `package.json`
- `yarn.lock`
- `ios/Podfile.lock`
- `src/entities/activity/ui/ActivityCard.tsx`
- `src/entities/unity/lib/hook/useUnityLifecycle.ts`
